pipeline/runtime/step_executor.go

```diff
@@ -276,10 +277,13 @@ func (e *StepExecutor) executeStep(r *api.StartStepRequest) (*runtime.State, map
 		result = multierror.Append(result, err)
 	}
 
-	// close the stream. If the session is a remote session, the
-	// full log buffer is uploaded to the remote server.
-	if err = wr.Close(); err != nil {
-		result = multierror.Append(result, err)
+	// if err is not nill or it's not a detach step then always close the stream
+	if err != nil || !r.Detach {
```

Comment:
Yeah so I have added the check `err != nil || !r.Detach` which basically means either error is not nil or it's not a detach step. This will ensure that stream is only closed here for non detach steps. We can't close the stream here for detach step since background step runs in a different routine and that is long running. If we close the stream here we'll lose the logs.

Reply:
got it.